File: strongtify_client/src/types/album.ts
import { Artist } from "./artist";
import { Genre } from "./genre";
import { PagingQuery } from "./paging";
import { Song } from "./song";

export type Album = {
    id: string;
    name: string;
    alias: string;
    imageUrl?: string;
    likeCount: number;
    songCount: number;
    totalLength: number;
    artist?: Artist;
}

export type AlbumDetail = {
    id: string;
    createdAt: string;
    name: string;
    alias: string;
    imageUrl?: string;
    likeCount: number;
    songCount: number;
    totalLength: number;
    artist?: Artist;
    genres?: Genre[];
    songs?: Song[];
}

export type CudAlbumResponse = {
    id: string;
    name: string;
    alias: string;
    imageUrl?: string;
    likeCount: number;
    songCount: number;
    totalLength: number;
    artistId?: string;
}

export type CreateUpdateAlbumRequest = {
    name: string;
    artistId?: string;
    genreIds?: string[];
    image?: File;
}

export type AlbumQuery = PagingQuery & {
    artistId?: string;
    genreId?: string;
}

File: strongtify_mobile_app/lib/models/album/album.dart
import 'package:strongtify_mobile_app/models/artist/artist.dart';

class Album {
  Album({
    required this.id,
    required this.name,
    required this.alias,
    this.imageUrl,
    this.likeCount = 0,
    this.artist,
  });

  final String id;
  final String name;
  final String alias;
  final String? imageUrl;
  final int likeCount;
  final Artist? artist;

  factory Album.fromMap(Map<String, dynamic> data) {
    return Album(
      id: data['id'],
      name: data['name'],
      alias: data['alias'],
      imageUrl: data['imageUrl'],
      likeCount: data['likeCount'] ?? 0,
      artist: data['artist'] != null ? Artist.fromMap(data['artist']) : null,
    );
  }
}
